src/rws_interface.cpp

```diff
@@ -276,14 +276,23 @@ RWSInterface::SystemInfo RWSInterface::getSystemInfo()
   SystemInfo result;
 
   RWSClient::RWSResult rws_result = rws_client_.getRobotWareSystem();
-  std::vector<Poco::XML::Node*> node_list = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_SYS_SYSTEM_LI);
 
+  std::vector<Poco::XML::Node*> node_list = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_SYS_SYSTEM_LI);
   for (size_t i = 0; i < node_list.size(); ++i)
   {
     result.system_name = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_NAME);
     result.robot_ware_version = xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_RW_VERSION_NAME);
   }
 
+  node_list = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_SYS_OPTION_LI);
+  for (size_t i = 0; i < node_list.size(); ++i)
+  {
+    result.system_options.push_back(xmlFindTextContent(node_list.at(i), XMLAttributes::CLASS_OPTION));
+  }
+
+  result.system_type = xmlFindTextContent(rws_client_.getContollerService().p_xml_document,
+                                          XMLAttributes::CLASS_CTRL_TYPE);
+
```

Comment:
can any of this fail? I'm not seeing any error handling.

Reply:
Clarified in 90ab7a9 that empty values will be returned in case of failure (e.g. communication timeout).